lib/hunter/commands/parse.rb

```diff
@@ -197,7 +197,7 @@ def manual_parse
 
           # Pre-generate the label, if possible
           prefill = answers[:active_choice].value.yield_self do |node|
-            node.preset_label || node.auto_label(used_names: @used_strings + reserved, default_prefix: @options.prefix, default_start: @options.start)
+            node.preset_label || node.auto_label(used_names: @used_strings + reserved, default_prefix: @options.prefix, default_start: @options.start, blank: @options.blank)
```

Comment:
I don't think it's the responsibility of the `Node#auto_label` method to implement this. It's the method's responsibility to generate a string to represent the object based on a set of options and variables. The occasion where the `blank` option is used means we shouldn't be attempting to generate any kind of string, which makes calling the `auto_label` method semantically pointless, and adds complexity to that method for no reason.

It's better implemented like this:

```ruby
# Labels from nodes that are currently selected in the parse menu
choices.each { |c| reserved << c[1].label }

# Evaluate blank option
@options.blank && prefix = ""

# Pre-generate the label, if possible
prefill ||= answers[:active_choice].value.yield_self do |node|
  node.preset_label || node.auto_label(used_names: @used_strings + reserved, default_prefix: @options.prefix, default_start: @options.start)
end

```

Reply:
My understanding is that `--blank` should only make the prefill blank if there aren't any better prefills to generate than the hostname - if a node has another way of finding its own label (e.g. a prefix/suffix combo) then that should still be the prefill. Have just clarified with @ColonelPanicks and that was the desired functionality of the option - I think the next steps for this PR are to merge the `--blank` option in this version with the `--short-hostname` config option such that we have a new option (available as a command option or config option) which allows you to select either `short` or `blank` to modify the prefill values for each node when parsing.

`auto_label` is the function that handles the prefill generation but we don't know if the resulting label was generated from the hostname or through other means, so the `--blank` functionality needs to be implemented inside of `auto-label` if the above

Reply:
> My understanding is that `--blank` should only make the prefill blank if there aren't any better prefills to generate than the hostname

I missed that in the spec; thanks for clarifying.

> I think the next steps for this PR are to merge the `--blank` option in this version with the `--short-hostname` config option such that we have a new option

Sounds good to me. Let me know when this is ready for another look.

Reply:
New changes have been implemented and the PR is ready for re-review @VoxSecundus

- The changes described here have been implemented:

> I think the next steps for this PR are to merge the `--blank` option in this version with the `--short-hostname` config option such that we have a new option (available as a command option or config option) which allows you to select either `short` or `blank` to modify the prefill values for each node when parsing.

- The field for the user to enter the new label has been made a little more verbose, only a minor change so I've included it with this PR